### ISSUE TITLE
Use of `LogAscii::logdir` collides with log rotation

### ISSUE THREAD

**Issue by ckreibich** [zeek/zeek#2120]
I hit this pretty much by accident ... in addition to using a `LogAscii::logdir`, enable `LogAscii::enable_leftover_log_rotation` and use a non-zero `rotation_interval`. The shadow file mechanism then breaks because the `.shadow.` and `.tmp` prefixes get tucked on before the `logdir`, and logger threads exit one by one:

```
$ zeek -C -i wlan0 LogAscii::logdir=foo/bar LogAscii::enable_leftover_log_rotation=T Log::default_rotation_interval=10sec
<params>, line 1: listening on wlan0

error in <params>, line 1: packet_filter/Log::WRITER_ASCII: cannot open .tmp.shadow.foo/bar/packet_filter.log: No such file or directory
error in <params>, line 1: packet_filter/Log::WRITER_ASCII: terminating thread
error in <params>, line 1: reporter/Log::WRITER_ASCII: cannot open .tmp.shadow.foo/bar/reporter.log: No such file or directory
error in <params>, line 1: reporter/Log::WRITER_ASCII: terminating thread
...
```
Seems this needs more precise path wrangling — those files should be `foo/bar/.tmp.shadow.packet_filter.log`, etc.

**Comment by awelzel** [comment]
> Seems this needs more precise path wrangling — those files should be foo/bar/.tmp.shadow.packet_filter.log, etc.

I looked a tiny bit. There might be a bit more to it. The [find_leftover_logs](https://github.com/zeek/zeek/blob/master/src/logging/writers/ascii/Ascii.cc#L738) function also needs updating to respect logdir rather than enumerating cwd for find the shadow files.

I'm happy to give this a go - I'm a tiny bit familiar with that area.

**Comment by awelzel** [comment]
Closed with #2223.
